src/fal/context.py

```diff
@@ -0,0 +1,107 @@
+# This file is auto-generated by tools/generate_typing_context.py, please
+# don't manually alter the contents.
+
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import pandas as pd
+    from faldbt.project import DbtModel, DbtTest, Feature
+    from fal.fal_script import Context
+    from typing import Optional, Dict, List, Protocol, Any
```

Comment:
The reason we use protocols here is the `Callable[]` syntax can't really represent some stuff (like defaults and doc-strings) https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols